txstratum/manager.py

```diff
@@ -133,8 +133,23 @@ def submit_solution(self, protocol: StratumProtocol, job: MinerJob, nonce: bytes
             assert isinstance(job, MinerTxJob)
             # Remove from queue.
             tx_job = job.tx_job
-            tx_job2 = self.tx_queue.popleft()
-            assert tx_job is tx_job2
+
+            if tx_job.status == JobStatus.DONE:
```

Comment:
What if the status is `TIMEOUT`, `FAILED`, or `CANCELLED`?

Reply:
I'll add the in the condition